environment-mac.yml

```diff
@@ -17,42 +20,39 @@ dependencies:
   # CONDA_SUBDIR=osx-arm64 conda env create -f environment-mac-updated.yml && conda list -n invokeai-updated | awk ' {print "  - " $1 "==" $2;} '
   # ```
 
-  - albumentations=1.2.1
-  - coloredlogs=15.0.1
-  - diffusers=0.6.0
-  - einops=0.4.1
-  - grpcio=1.46.4
+  - albumentations=1.2
+  - coloredlogs=15.0
+  - diffusers=0.6
+  - einops=0.3
+  - eventlet
+  - grpcio=1.46
+  - flask=2.1
+  - flask-socketio=5.3
+  - flask-cors=3.0
   - humanfriendly=10.0
-  - imageio=2.21.2
-  - imageio-ffmpeg=0.4.7
-  - imgaug=0.4.0
-  - kornia=0.6.7
-  - mpmath=1.2.1
-  - nomkl # arm64 has only 1.0 while x64 needs 3.0
-  - numpy=1.23.4
-  - omegaconf=2.1.1
-  - openh264=2.3.0
-  - onnx=1.12.0
-  - onnxruntime=1.12.1
-  - pudb=2022.1
-  - pytorch-lightning=1.7.7
-  - scipy=1.9.3
-  - streamlit=1.12.2
-  - sympy=1.10.1
-  - tensorboard=2.10.0
-  - torchmetrics=0.10.1
-  - py-opencv=4.6.0
-  - flask=2.1.3
-  - flask-socketio=5.3.0
-  - flask-cors=3.0.10
-  - eventlet=0.33.1
-  - protobuf=3.20.1
-  - send2trash=1.8.0
-  - transformers=4.23.1
-  - torch-fidelity=0.3.0
+  - imageio=2.21
+  - imageio-ffmpeg=0.4
+  - imgaug=0.4
+  - kornia=0.6
+  - mpmath=1.2
+  - nomkl=3
+  - numpy=1.23
+  - omegaconf=2.1
+  - openh264=2.3
+  - onnx=1.12
+  - onnxruntime=1.12
+  - pudb=2019.2
+  - protobuf=3.20
+  - py-opencv=4.6
+  - scipy=1.9
+  - streamlit=1.12
+  - sympy=1.10
+  - send2trash=1.8
+  - tensorboard=2.10
+  - transformers=4.23
   - pip:
-      - getpass_asterisk
       - dependency_injector==4.40.0
+      - getpass_asterisk
       - realesrgan==0.2.5.0
       - test-tube==0.7.5
       - -e git+https://github.com/CompVis/taming-transformers.git@master#egg=taming-transformers
```

Comment:
I've found that we can get rid of the last `src` dependency by replacing this line with `- taming-transformers-rom1504`, which seems to be a drop-in replacement. Could you see if that's possible for Macs?

Reply:
In addition, for the following 4 lines I think we can now get rid of the `-e` in order to have the modules installed in the user's environment rather than kept in the InvokeAI `src` directory.

I've done this systematically for the Linux and Windows environment files, and it seems to work like a charm. Not tested on OSX due to lack of access to an appropriate system.

Reply:
I do not understand how the github action step to `python scripts/preload_models.py --no-interactive` is working, while localy I get this error now:

```
Loading Python libraries...

Traceback (most recent call last):
  File "/Users/mauwii/git/invoke-ai/InvokeAI/scripts/preload_models.py", line 19, in <module>
    from getpass_asterisk import getpass_asterisk
ModuleNotFoundError: No module named 'getpass_asterisk'
```

I even did a `git clean -ixd` and a `git reset --hard update-environment-mac` to make sure I have the same state which the action should have, so maybe the git runners have this package (getpass_asterisk) preinstalled 🤔

Reply:
Ah - lol - sorry, I was looking at the wrong run (where I merged development into update-environment-mac), after changing the requirements, the action is failing with the same error now...

So I will put `getpass_asterisk` back in instead of `taming-transformers-rom1504`

BTW: You have access to a MacOS System, kind of, with the actions 😅

Reply:
I was a bit irritated by the highlighted line 🤦🏻 

<img width="788" alt="image" src="https://user-images.githubusercontent.com/40327258/200646610-738ba1f6-0a15-4d9e-98ee-cc65a9b5943c.png">

Think I nailed it now, just want to se the result of the action before "resolving the conversation" 🤓